roles/kafka_controller/tasks/main.yml

```diff
@@ -0,0 +1,460 @@
+  retries: 5
+  delay: 90
+  loop: "{{kafka_controller_packages}}"
+  when:
+    - ansible_os_family == "RedHat"
+    - installation_method == "package"
+  ignore_errors: "{{ ansible_check_mode }}"
+  tags: package
+  notify: restart kafka
+
+- name: Install the Kafka Controller Packages
+  apt:
+    name: "{{ kafka_controller_packages | product([confluent_package_debian_suffix]) | map('join') | list }}"
+  register: install_kafka_package_result
+  until: install_kafka_package_result is success  or ansible_check_mode
+  retries: 5
+  delay: 90
+  loop: "{{kafka_controller_packages}}"
+  when:
+    - ansible_os_family == "Debian"
+    - installation_method == "package"
+  ignore_errors: "{{ ansible_check_mode }}"
+  tags: package
+  notify: restart kafka
+
+- name: Kafka Controller group
+  group:
+    name: "{{kafka_controller_group}}"
+  tags:
+    - privileged
+
+- name: Check if Kafka Controller User Exists
+  # Create user task must be skipped for non-linux users
+  getent:
+    database: passwd
+    key: "{{kafka_controller_user}}"
+  failed_when: false
+
+- name: Create Kafka Controller user
+  user:
+    name: "{{kafka_controller_user}}"
+    comment: Confluent Kafka
+    system: true
+    shell: "{{user_login_shell}}"
+    group: "{{kafka_controller_group}}"
+  when: (getent_passwd|default({}))[kafka_controller_user] is not defined
+  tags:
+    - privileged
+
+# Archive File deployments need to create SystemD service units
+# Copy the tarball's systemd service to the system
+- name: Copy Kafka Controller Service from archive file to system
+  copy:
+    src: "{{binary_base_path}}/lib/systemd/system/{{kafka_controller.systemd_file|basename}}"
+    remote_src: true
+    dest: "{{kafka_controller.systemd_file}}"
+    mode: 0644
+    force: true
+  when: installation_method == "archive"
+  tags:
+    - systemd
+
+- include_role:
+    name: ssl
+  vars:
+    truststore_storepass: "{{kafka_controller_truststore_storepass}}"
+    truststore_path: "{{kafka_controller_pkcs12_truststore_path}}"
+    keystore_path: "{{kafka_controller_pkcs12_keystore_path}}"
+    keystore_storepass: "{{kafka_controller_keystore_storepass}}"
+    keystore_keypass: "{{kafka_controller_keystore_keypass}}"
+    service_name: kafka_controller
+    user: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+    hostnames: "{{ ([inventory_hostname, ansible_fqdn, ansible_host, ansible_ssh_host|default(omit), hostname|default(omit)] + kafka_controller_listeners | confluent.platform.get_hostnames(inventory_hostname)) | unique }}"
+    ca_cert_path: "{{kafka_controller_ca_cert_path}}"
+    cert_path: "{{kafka_controller_cert_path}}"
+    key_path: "{{kafka_controller_key_path}}"
+    export_certs: "{{kafka_controller_export_certs}}"
+    create_bouncy_castle_keystore: "{{fips_enabled}}"
+    bcfks_truststore_path: "{{kafka_controller_bcfks_truststore_path}}"
+    bcfks_keystore_path: "{{kafka_controller_bcfks_keystore_path}}"
+  when: >
```

Comment:
Should we add a check `kafka_controller_ssl_enabled` as well?
If we do, we have to make sure that the check is superseded by `fips_enabled`, meaning don't adhere this flag if fips are enabled.